scipy/special/wright.cc

```diff
@@ -0,0 +1,389 @@
+    {
+      pz=conj(sqrt(conj(2.0*(z+1.0-I*pi))));
+      *w=-1.0+(I+(1.0/3.0+(-1.0/36.0*I+(1.0/270.0+1.0/4320.0*I*pz)*pz)*pz)*pz)*pz;
+    }
+  /**********************************/
+  /* Region 2: lower branch point   */
+  /* Series about z=-1-Pi*I         */
+  /**********************************/
+  else if ((-2.0<x && x<=1.0 && -2.0*pi<y && y<-1.0))
+    {
+      pz=conj(sqrt(conj(2.0*(z+1.0+I*pi))));
+      *w=-1.0+(-I+(1.0/3.0+(1.0/36.0*I+(1.0/270.0-1.0/4320.0*I*pz)*pz)*pz)*pz)*pz;
+    }
+  /*********************************/
+  /* Region 3: between branch cuts */
+  /* Series: About -infinity       */
+  /*********************************/
+  else if (x <= -2.0 && -pi < y && y <= pi)
+    {
+      pz=exp(z);
+      *w=(1.0+(-1.0+(3.0/2.0+(-8.0/3.0+125.0/24.0*pz)*pz)*pz)*pz)*pz;
+      if (*w == 0.0)
+	{
+	  sf_error("wrightomega", SF_ERROR_UNDERFLOW, "underflow in exponential series");
+	  /* Skip the iterative scheme because it computes log(*w) */
+	  e = NULL;
+	  r = NULL;
+	  if (cond != NULL)
+	    {
+	      *cond = z/(1.0+*w);
+	    }
+	  return 0;
+	}
+    }
+  /**********************/
+  /* Region 4: Mushroom */
+  /* Series about z=1   */
+  /**********************/
+  else if (((-2.0 < x) && (x<=1.0) && (-1.0 <= y) && (y <= 1.0))
+           || ((-2.0 < x) && (x-0.1e1)*(x-0.1e1)+y*y<=pi*pi))
+    {
+      pz=z-1.0;
+      *w=1.0/2.0+1.0/2.0*z+(1.0/16.0+(-1.0/192.0+(-1.0/3072.0+13.0/61440.0*pz)*pz)*pz)*pz*pz;
+    }
+  /*************************/
+  /* Region 5: Top wing    */
+  /* Negative log series   */
+  /*************************/
+  else if (x<=-0.105e1 && pi<y && y-pi<=-0.75e0*(x+0.1e1))
+    {
+      t=z-I*pi;
+      pz=log(-t);
+      *w = t - pz;
+      fac = pz/t;
+      *w += fac;
+      fac /= t;
+      *w += fac*(0.5*pz - 1.0);
+      fac /= t;
+      *w += fac*(pz*pz/3.0 - 3.0*pz/2.0 + 1.0);
+      if (abs(z) > 1e50)
+	/* Series is accurate and the iterative scheme could overflow */
+	{
+	  e = NULL;
+	  r = NULL;
+	  if (cond != NULL)
+	    {
+	      *cond = z/(1.0+*w);
+	    }
+	  return 0;
+	}
+    }
+  /***************************/
+  /* Region 6: Bottom wing   */
+  /* Negative log series     */
+  /***************************/
+  else if (x<=-0.105e1 && 0.75e0*(x+0.1e1)< y+pi && y+pi<=0.0e0)
+    {
+      t=z+I*pi;
+      pz=log(-t);
+      t=z-I*pi;
+      pz=log(-t);
+      *w = t - pz;
+      fac = pz/t;
+      *w += fac;
+      fac /= t;
+      *w += fac*(0.5*pz - 1.0);
+      fac /= t;
+      *w += fac*(pz*pz/3.0 - 3.0*pz/2.0 + 1.0);
+      if (abs(z) > 1e50)
+	/* Series is accurate and the iterative scheme could overflow */
+	{
+	  e = NULL;
+	  r = NULL;
+	  if (cond != NULL)
+	    {
+	      *cond = z/(1.0+*w);
+	    }
+	  return 0;
+	}
+    }
+  /************************************/
+  /* Region 7: Everywhere else        */
+  /* Series solution about infinity   */
+  /************************************/
+  else
+    {
+      pz=log(z);
+      *w = z - pz;
+      fac = pz/z;
```

Comment:
They should be covered in the `TestSystematic` test--if you go back to https://github.com/scipy/scipy/pull/6653/commits/347d7d384585ad8f4046455bdb80faa31bc0d73e but flip the `nan_ok` flag in `TestSystematic.test_wrightomega` to `False` you'll get around 1,200 failures, which these changes are trying to address.